### ISSUE TITLE
Detect element resize

### ISSUE THREAD

**Issue by VictorJuliani** [Komock/ngx-skltn#1]
Hi there!

I stumbled upon this library and really liked the simplicity and flexibility of the solution!
My issue is more a discussion then a problem.

I am creating a page that may have components resized (for example a sidebar menu being opened/closed) after calcShapes has finished running. Because of that, the SVG dimensions become wrong and skeletons get distorted/mispositioned.
As an alternative, I've tried to use ngDoCheck and to perform calcShapes when dimensions change, but this depends on change detection being run. A parent component using OnPush strategy will make this solution unhelpful.
Do you have any other idea on how to work around this problem?
I've considered using an interval observable to perform this checks, but I'm not sure if this is the best idea in a scenario with several skeletons. Perhaps a shared observable declared on SkltnService and subscribed on each SkltnComponent.

Also, is there any reason for the component not use OnPush Strategy?
And finally, shouldn't `updStream$` stream be unsubscribed on ngOnDestroy? I belive there will be a leak if no unsubscribe is made.

**Comment by Komock** [comment]
Hi!
As quick solution you can dispatch window resize:
[https://stackblitz.com/edit/ngx-skltn-bzuhlv](https://stackblitz.com/edit/ngx-skltn-bzuhlv)

Later i'll add possibility to run calcShapes outside component.

OnPush strategy can be inherit from parent component.
And yes, i'll add unsubscribe, thanks!

**Comment by VictorJuliani** [comment]
@Komock Thanks for you reply!

I wanted to create an independent solution, as I have no guarantees that a component resize can be caught in order to trigger the window resize event. I've tried to add the interval solution, but even in 100ms the result isn't fluid, though it works as expected.

I found a library called angular-resize-event (https://github.com/vdolek/angular-resize-event), but I believe it works through polling and this may not be a good solution, should a page have many items being listened for resize.
